common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -318,9 +318,46 @@ private static Collation fetchCollation(int collationId) {
         }
       }
 
-      protected static SparkException collationInvalidNameException(String collationName) {
+      protected static SparkException collationInvalidNameException(
+          String collationName,
+          String[] ICULocaleNames,
```

Comment:
Since this function combines code path for both ICU collations and `UTF8_BINARY`/`UTF8_LCASE`, please consider renaming this parameter.